Tools/cases_generator/parser.py

```diff
@@ -278,14 +322,48 @@ def stack_effect(self) -> StackEffect | None:
                 cond_text = cond.text.strip()
             size_text = ""
             if self.expect(lx.LBRACKET):
-                if type_text or cond_text:
+                if has_type_annotation or cond_text:
                     raise self.make_syntax_error("Unexpected [")
                 if not (size := self.expression()):
                     raise self.make_syntax_error("Expected expression")
                 self.require(lx.RBRACKET)
-                type_text = "PyObject **"
+                type = "PyObject **"
```

Comment:
This assignment shadows a builtin. According to PEP 8, we need to change below too.
```suggestion
                type_ = "PyObject **"
```